src/fundus/scraping/crawler.py

```diff
@@ -106,10 +113,213 @@ def pool_queue_iter(handle: MapResult[Any], queue: Queue[_T]) -> Iterator[_T]:
+                with None values for respective attributes in the extraction or by skipping entire articles.
+                If set to "catch", errors will be caught as attribute values or, if an entire article fails,
+                through Article.exception. If set to "raise" all errors encountered during extraction will
+                be raised. Defaults to "suppress".
+            only_complete (Union[bool, ExtractionFilter]): Set a callable satisfying the ExtractionFilter
+                protocol as an extraction filter or use a boolean. If False, all articles will be yielded,
+                if True, only those with all attributes extracted. Defaults to ExtractionFilter letting
+                through all articles with at least title, body, and publishing_date set.
+            url_filter (Optional[URLFilter]): A callable object satisfying the URLFilter protocol to skip
+                URLs before download. This filter applies on both requested and responded URL. Defaults to None.
+            only_unique (bool): If set to True, articles yielded will be unique on the responded URL.
+                Always returns the first encountered article. Defaults to True.
+
+        Returns:
+            Iterator[Article]: An iterator yielding objects of type Article.
+        """
+
+        if max_articles == 0:
+            return
+
+        if max_articles is None:
+            max_articles = -1
+
+        def build_extraction_filter() -> Optional[ExtractionFilter]:
+            if isinstance(only_complete, bool):
+                return None if only_complete is False else RequiresAll()
+            else:
+                return only_complete
+
+        response_cache: Set[str] = set()
+
+        extraction_filter = build_extraction_filter()
+        fitting_publisher: List[PublisherEnum] = []
+
+        if isinstance(extraction_filter, Requires):
+            for publisher in self.publishers:
+                supported_attributes = set(
+                    more_itertools.flatten(
+                        collection.names for collection in publisher.parser.attribute_mapping.values()
+                    )
+                )
+                if missing_attributes := extraction_filter.required_attributes - supported_attributes:
+                    basic_logger.warning(
+                        f"The required attribute(s) `{', '.join(missing_attributes)}` "
+                        f"is(are) not supported by {publisher.publisher_name}. Skipping publisher"
+                    )
+                else:
+                    fitting_publisher.append(publisher)
+
+            if not fitting_publisher:
+                basic_logger.error(
+                    f"Could not find any fitting publisher for required attributes  "
+                    f"`{', '.join(extraction_filter.required_attributes)}`"
+                )
+                return
+
+        article_idx = 0
```

Comment:
You're right! Something else: Suggesting code changes here within GitHub seems somewhat dangerous because it's tempting to just commit then but they won't propagate to the rest of the code.

Reply:
Good point, I'll keep that in mind